details/input.md

```diff
@@ -0,0 +1,172 @@
+authenticated data. Event data varies depending on whether it is a source or a trigger event. Authenticated data is used
+to decrypt the shares of match key.
+
+Two things that consume the most space on the wire are **site registrable domain** and **match key provider origin**. 
+Both are ASCII strings, potentially large, that each event must refer to in order for helper parties to correctly obtain 
+the plain text shares.
+
+The biggest savings from the custom format come from making each query to carry only one copy of unique site domain
+and match key provider origin strings. This proposal suggests building two lookup tables (one for each entity) on the 
+caller site
+
+
+## Proposed format
+
+Each query request must carry the following information: 
+```text
+[match key][event data][authenticated data][match key][event data][authenticated data]...
+```
+
+This proposal suggests splitting query requests into two sections: lookup tables section for match key providers and 
+site domains and payload section with encrypted replicated shares of match key and event data. Each encryption is annotated
+with a unique id pair that points to site and match key provider strings that are used to authenticate the match key
+encryption.
+
+```text
+.lookups
+[site origin 1][site origin 2]..[site origin N]
+[match key provider 1][match key provider 2]..[match key provider M]
+.payload
+[site origin id][match key provider id][authenticated data][match key][event data]
+[site origin id][match key provider id][authenticated data][match key][event data]
+...
+```
+
+where `id` is the index of site origin or match key provider inside the lookup table.
+
+It is natural to assume $M \ll N$, so fewer bits is required to encode match key provider index. 
+
+The total number of site domain entries inside the lookup table must be less than $2^{32}$.
+The total number of match key provider origin entries inside the lookup table must be less than $2^8$. 
+
+
+### Metadata
+
+Every query request must specify several parameters to the helper parties that impact the size of the payload.
+These parameters are sent in the [header](https://www.rfc-editor.org/rfc/rfc9110.html#name-header-fields) of HTTP request.
+
+The list of supported parameters include:
+
+| Header name      | Type                         | Description                            | Accepted values | Default? | Mandatory? |
+|------------------|------------------------------|----------------------------------------|-----------------|----------|------------| 
+| `x-ipa-field`    | US-ASCII encoded string      | Field type used to secret-share values | `fp32`          | No       | Yes        |
+| `x-ipa-query`    | US-ASCII encoded string      | Desired query to run in MPC            | `ipa`           | `ipa`    | No         |
+| `x-ipa-version`  | single byte unsigned integer | Version of the request                 | `1`             | No       | Yes        |
```

Comment:
RFC 6648.

We need a format for creating a query, which needs to include these values somehow.  I would not use header fields for this, but instead define a payload format.  This doesn't need to be tightly packed, so JSON is probably where I would go.  

Also, some of this is information that could be part of the resource identity.  That is, you would have one URI that does IPA and another that does something different.  That means that you don't need to include explicit versioning.

Parameters are only necessary if you think that something needs tuning, or there are things that need to be known in order to accept the query.  I think that we should directly signal the query size in this request as that has a direct bearing on what is being requested.  

IPA already has a bunch of parameters that we have built into our implementation:

* The number of breakdown keys.
* The maximum value of individual trigger values.
* The per-user cap.
* The attribution window.

These are what I would expect to see in the request that creates a query.